File: RepositoryOld/MessageRepository.fs
module Repository.MessageRepository

open Repository.HarbingerContext
open Repository.Message
open Repository.RepositoryInterfaces

type MessageRepository(harbingerContext: HarbingerContext) =
    member this.HarbingerContex = harbingerContext
    
    interface IRepository<Message> with
        member this.Create(message) =
            async {
                this.HarbingerContex.messages.AddAsync(message) |> Async.AwaitTask |> ignore
                this.HarbingerContex.SaveChangesAsync() |> Async.AwaitTask |>ignore
                return message
            }
        

File: Repository/MessageRepository.cs
using AutoMapper;
using Domain;

namespace Repository
{
    public class MessageRepository : IRepository<Message.Message>
    {
        private HarbingerContext _harbingerContext;
        private IMapper _mapper;

        public MessageRepository(HarbingerContext harbingerContext, IMapper mapper)
        {
            _harbingerContext = harbingerContext;
            _mapper = mapper;
        }
        
        public Message.Message Create(Message.Message message)
        {
            var messageDto = _mapper.Map<Message.Message, MessageDto>(message);
            _harbingerContext.Messages.Add(messageDto);
            _harbingerContext.SaveChanges();
            return _mapper.Map<MessageDto, Message.Message>(messageDto);
        }
    }
}